src/test/compile-fail/borrowck/borrowck-overloaded-index-and-overloaded-deref.rs

```diff
@@ -39,7 +42,9 @@ fn main() {
     let mut v = MyVec { x: MyPtr { x: Foo { f: 22 } } };
     let i = &v[0].f;
     v = MyVec { x: MyPtr { x: Foo { f: 23 } } };
-    //~^ ERROR cannot assign to `v`
+    //[ast]~^ ERROR cannot assign to `v`
+    //[mir]~^^ ERROR cannot assign to `v` because it is borrowed (Ast)
```

Comment:
the `~^^` here could just be `~|`

Reply:
Cf. https://github.com/rust-lang/rust/pull/44811#discussion_r141350517